### ISSUE TITLE
Throw an error when increment or decrement signs are in the prefix of a number in the string interpolation expression.

### ISSUE THREAD

**Issue by yassine-klilich** [angular/angular#35779]
# 🚀 feature request

### Relevant Package
This feature request is for @angular/compiler

### Description
I noticed if this string interpolation expression {{num++}} throws an error because we can't do incrementalism in a interpolation expression, so I see that also that incrementalism with this interpolation expression {{++num}} should throw an error. The same thing with decrement of a number, {{num--}} throws an error, but {{--num}} does not.
I made a quick example here : https://stackblitz.com/edit/angular-fkehrc?file=src%2Fapp%2Fapp.component.html

### Describe the solution you'd like
The solution is in both cases the string interpolation expression {{num++}} and {{++num}} should throw an error.

**Comment by pkozlowski-opensource** [comment]
Wow, you are totally right, this is not consistent at all :-)
For the reference we've got a related issue #32882 (I guess for both cases we should have a more descriptive message, something around an expression not being supported).

**Comment by pkozlowski-opensource** [comment]
I'm going to close it as a duplicate of #32882 based on my comment in there: https://github.com/angular/angular/issues/32882#issuecomment-626681317

tl;dr; is that I think we should support it or, as the very least, consistently report errors during compilation time.
